### ISSUE TITLE
How to fill future known information from one of the exogenous variables?

### ISSUE THREAD

**Issue by kennis222** [JoaquinAmatRodrigo/skforecast#314]
Hi developers,

I have a question about fill future known information from one of the exogenous variables.
For example, I have a data where is y as target variable and three exogenous variables X1,X2, and X3. If I have future known information of X3, but no future information of X1,X2. Either using directed or recursive methods, how to integrate the future known exogenous variables (X3) and without future known (X1,X2) and integrate this in the designed framework package?

**Comment by JavierEscobarOrtiz** [comment]
Hello @kennis222,

From my point of view there are 2 options:

1. Since an exogenous variable is one whose measure is determined outside the model and is imposed on the model, you can create a `ForecasterAutoreg` to predict the future for X1 and X2 and then use the predictions to predict your target variable. Here the main problem is that you are including the forecast error of your X1 and X2 variables in your final model.

2. Another option is to use ' ForecasterAutoregMultiVariate`. Here you can treat X1 and X2 as multivariate variables and only use 1 lag. With this approach, the future of X1 and X2 is not necessary since it is a direct method.

In this example, I use 7 lags from `y` and only lag 1 from `X1` and `X2`:

```python
# Create and fit forecaster MultiVariate Custom lags
# ==============================================================================
forecaster = ForecasterAutoregMultiVariate(
                 regressor          = Ridge(random_state=123),
                 level              = 'y',
                 lags               = {'y': 7, 'X1': 1, 'X2': 1},
                 steps              = 7,
                 transformer_series = None,
                 transformer_exog   = None,
                 weight_func        = None
             )

forecaster.fit(series=data[['y', 'X1', 'X2']], exog=data[['X3']])

# Predict MultiVariate
# ==============================================================================
predictions = forecaster.predict(steps=7, exog=data_future[['X3']])
```

Hope it helps!

**Comment by kennis222** [comment]
Hi developers, 

I tried to use this ForecasterAutoregMultiVariate but combining with the grid_search_forecaster_multiseries to tune the fitted  training model. However, I encounter the problem of lags from different variables. Continue the example above, we can set the different lags from ForecasterAutoregMultiVariate for different variables. However, when combing with grid_search_forecaster_multiseries, the lags setting from ForecasterAutoregMultiVariate was ignored. Instead, the **lags_grid** from grid search dominates. The problem is,  lags_grid from grid search cannot specify the lags from different variables.  

For example, if I set lags_grid = [1,2,3] then y_lag1,y_lag2,y_lag3, X1_lag1,X1_lag2,X1_lag3,X2_lag1,X2_lag2, and X2_lag3. What I want in the grid_search is specify the lags for different variables, in other words, y_lag1,y_lag2,y_lag3, X1_lag2, and X2_lag3, which is I want in my real situation. How to conduct in the grid_search?

Regards.
